lib/ruby_llm/tool.rb

```diff
@@ -49,14 +49,14 @@ def parameters
     end
 
     def name
-      self.class.name
-          .unicode_normalize(:nfkd)
-          .encode('ASCII', replace: '')
-          .gsub(/[^a-zA-Z0-9_-]/, '-')
-          .gsub(/([A-Z]+)([A-Z][a-z])/, '\1_\2')
-          .gsub(/([a-z\d])([A-Z])/, '\1_\2')
-          .downcase
-          .delete_suffix('_tool')
+      klass_name = self.class.name
+      normalized = klass_name.ascii_only? ? klass_name : klass_name.unicode_normalize(:nfkd)
+      normalized.encode('ASCII', replace: '')
```

Comment:
This assumes the encoding is going to be either ASCII or Unicode. Is there a way to convert any encoding to unicode instead?